crates/flux-refineck/src/checker.rs

```diff
@@ -1039,6 +1067,142 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
         }
     }
 
+    fn closure_template(
+        &mut self,
+        infcx: &mut InferCtxt<'_, 'genv, 'tcx>,
+        env: &mut TypeEnv,
+        stmt_span: Span,
+        args: &flux_rustc_bridge::ty::GenericArgs,
+        operands: &[Operand],
+    ) -> Result<(Vec<Ty>, PolyFnSig)> {
```

Comment:
You could return an `InferResult` here to avoid having to use `with_span`. Then you only have a `with_span` when you call `closure_template`.

```suggestion
    ) -> InferResult<(Vec<Ty>, PolyFnSig)> {
```